python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -1261,7 +1261,7 @@ def recover_if_needed(self, node_id, now):
             process_runner=self.process_runner,
             use_internal_ip=True,
             is_head_node=False,
-            docker_config=self.config.get("docker"),
+            docker_config=self._get_node_specific_docker_config(node_id),
```

Comment:
> @rueian could you review this one?

Sure. Hi @JohnsonKuan, the change of using `_get_node_specific_docker_config`  looks good to me.
```python
    def _get_node_specific_docker_config(self, node_id):
        if "docker" not in self.config:
            return {}
        docker_config = copy.deepcopy(self.config.get("docker", {}))
        node_specific_docker = self._get_node_type_specific_fields(node_id, "docker")
        docker_config.update(node_specific_docker)
        return docker_config
```
I think using `_get_node_specific_docker_config` is correct and pretty safe. However, can we still have a test in `test_autoscaler.py`? I think we could use `runner.assert_has_call` or `runner.command_history` to test which image it uses.

Reply:
Thanks. LGTM!

Reply:
Hi @jjyao, this PR is ready to be merged! PTAL.